### PR TITLE
feat: key-value pairs lint rule

### PR DESCRIPTION
**Pull request by adthrasher**
This pull request adds a new rule to `wdl-lint`.

- **Rule Name**: `KeyValuePiars`

This rule checks objects and arrays in `meta` and `parameter_meta` sections. It reports any that are on a single line. If already on multiple lines, it checks for missing newlines between elements. It also checks for indentation of each element and the closing delimiter.

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added an entry to the relevant `CHANGELOG.md` (see
      ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

Rule specific checks:

- [x] You have added the rule as an entry within `RULES.md`.
- [x] You have added the rule to the `rules()` function in `wdl-lint/src/lib.rs`.
- [x] You have added a test case in `wdl-lint/tests/lints` that covers every
      possible diagnostic emitted for the rule within the file where the rule
      is implemented.
- [ ] If you have implemented a new `Visitor` callback, you have also
      overridden that callback method for the special `Validator`
      (`wdl-ast/src/validation.rs`) and `LintVisitor`
      (`wdl-lint/src/visitor.rs`) visitors. These are required to ensure the new
      visitor callback will execute.
- [x] You have run `wdl-gauntlet --refresh` to ensure that there are no 
      unintended changes to the baseline configuration file (`Gauntlet.toml`).
- [x] You have run `wdl-gauntlet --refresh --arena` to ensure that all of the 
      rules added/removed are now reflected in the baseline configuration file 
      (`Arena.toml`).

### CODE REVIEW

File: wdl-lint/RULES.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ be out of sync with released packages.
 | `InconsistentNewlines`           | Clarity, Style                | Ensures that newlines are used consistently within the file.                          |
 | `InputNotSorted`                 | Style                         | Ensures that input declarations are sorted                                            |
 | `InputSpacing`                   | Style, Clarity, Spacing       | Ensures proper spacing for call inputs                                                |
+| `KeyValuePairs`                  | Style                         | Ensures that metadata lists and arrays are properly spaced.                           |
```
Comment (a-frantz) [inline wdl-lint/RULES.md:25]:
```suggestion
| `KeyValuePairs`                  | Style                         | Ensures that metadata objects and arrays are properly spaced.                           |
```

File: wdl-lint/tests/lints/key-value-pairs/source.errors [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+9 │ │         "value2", "value3"]
+  │ ╰────────^
+  │  
+  = fix: add 4 spaces to indentation
+
+note[KeyValuePairs]: item should be followed by a newline
+  ┌─ tests/lints/key-value-pairs/source.wdl:9:9
+  │
+9 │         "value2", "value3"]
+  │         ^^^^^^^^^^
```
Comment (a-frantz) [inline wdl-lint/tests/lints/key-value-pairs/source.errors:23]: I don't think this reported span should include this whitespace

File: wdl-lint/tests/lints/key-value-pairs/source.errors [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+   │
+20 │                 "m", "n"],
+   │                 ^^^^^
+   │
+   = fix: add a newline after this element
+
+note[KeyValuePairs]: item should be followed by a newline
+   ┌─ tests/lints/key-value-pairs/source.wdl:20:22
+   │
+20 │                 "m", "n"],
```
Comment (a-frantz) [inline wdl-lint/tests/lints/key-value-pairs/source.errors:94]: It should also be followed by a comma. I just peeked lower and see you've `except`ed `TrailingComma`. Can you allow that rule?

Comment (adthrasher) [reply wdl-lint/tests/lints/key-value-pairs/source.errors:94]: I was trying to suppress errors that weren't relevant to this rule. If I remove the exception, there are additional lints.
```
+note[TrailingComma]: item missing trailing comma
+  ┌─ tests/lints/key-value-pairs/source.wdl:9:19
+  │
+9 │         "value2", "value3"]
+  │                   ^^^^^^^^
+  │
+  = fix: add a comma after this element
```

and 

```
+note[TrailingComma]: item missing trailing comma
+   ┌─ tests/lints/key-value-pairs/source.wdl:20:22
+   │
+20 │                 "m", "n"],
+   │                      ^^^
+   │
+   = fix: add a comma after this element
+
```

and 

```
+note[TrailingComma]: item missing trailing comma
+   ┌─ tests/lints/key-value-pairs/source.wdl:34:13
+   │
+34 │             choices: ["yes", "reverse", "no"]
+   │             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │
+   = fix: add a comma after this element
```

File: wdl-lint/tests/lints/key-value-pairs/source.errors (hunk 2 above)
Comment (adthrasher) [reply wdl-lint/tests/lints/key-value-pairs/source.errors]: OK, I made `find_next_comma()` a public crate function in the `TrailingComma` rule and imported it here. I am now using that to find the comma and running the `Span` until the end of the comma.

**Review by a-frantz — APPROVED** [review]

**Pull request merged**